<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">  
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    
    <title>Abinitio Annotation - Bioinformatics course</title>
    

    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/3.003/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="../../../css/base.min.css" rel="stylesheet">
    <link href="../../../css/cinder.min.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.min.css">
    <link href="../../../css/admonition.css" rel="stylesheet">
    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
            <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
        <![endif]-->

    <script src="//ajax.googleapis.com/ajax/libs/webfont/1.6.28/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    

     
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">First week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../blast/blast_online/">Blast Online</a>
</li>

                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../blast/blast_cli/">Command-line Blast</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Second week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/rna/">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/wms/">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 5 - Eukaryotic Annotation <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../schedule/">Schedule</a>
</li>

                        
                            
<li >
    <a href="../practical1/">Practical 1</a>
</li>

                        
                            
<li >
    <a href="../practical2/">Practical 2</a>
</li>

                        
                            
<li >
    <a href="../practical3_manualCuration/">Practical 3</a>
</li>

                        
                            
<li >
    <a href="../practical4_funcAnnotInterp/">Practical 4</a>
</li>

                        
                            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Extras</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../UsingWebApollo.md">WebApollo</a>
</li>

        
            
<li >
    <a href="../practical2_sub_gatherEvidence/">Protein, EST and RNA-seq data</a>
</li>

        
            
<li >
    <a href="../practical2_sub_transcriptome/">Assembling transcripts</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerNoAbinit/">Evidence Based Annotation</a>
</li>

        
            
<li class="active">
    <a href="./">Abinitio Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerCompareAnnot/">Comparing Annotations</a>
</li>

        
            
<li >
    <a href="../practical2_supl_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl3_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl2_maker/">Inspecting Maker output</a>
</li>

        
    </ul>
  </li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fas fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../practical2_sub_makerNoAbinit/">
                            <i class="fas fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../practical2_sub_makerCompareAnnot/">
                            Next <i class="fas fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#making-an-abinitio-evidence-driven-annotation-with-maker">Making an abinitio evidence-driven annotation with MAKER</a></li>
            <li class="second-level"><a href="#prepare-the-input-data">Prepare the input data</a></li>
                
            <li class="second-level"><a href="#run-maker-with-ab-initio-predictions">Run Maker with ab-initio predictions</a></li>
                
            <li class="second-level"><a href="#compile-the-output">Compile the output</a></li>
                
            <li class="second-level"><a href="#inspect-the-gene-models">Inspect the gene models</a></li>
                
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="making-an-abinitio-evidence-driven-annotation-with-maker">Making an abinitio evidence-driven annotation with MAKER</h1>
<p>The recommended way of running Maker is in combination with one or more ab-initio profile models. Maker natively supports input from several tools, including augustus, snap and genemark. The choice of tool depends a bit on the organism that you are annotating - for example, GeneMark-ES is mostly recommended for fungi, whereas augustus and snap have a more general use.</p>
<p>The biggest problem with ab-initio models is the process of training them. It is usually recommended to have somewhere around 500-1000 curated gene models for this purpose. Naturally, this is a bit of a contradiction for a not-yet annotated genome.</p>
<p>However, if one or more good ab-initio profiles are available, they can potentially greatly enhance the quality of an annotation by filling in the blanks left by missing evidence. Interestingly, Maker even works with ab-initio profiles from somewhat distantly related species since it can create so-called hints from the evidence alignments, which the gene predictor can take into account to fine-tune the predictions.</p>
<p>Usually when no close ab-initio profile exists for the investigated species, we use the first round of annotation (evidence based) to create one. We first filter the best gene models from this annotation, which are used then to train the abinitio tools of our choice.</p>
<p>In order to compare the performance of Maker with and without ab-initio predictions in a real-world scenario, we have first run a gene build without ab-initio predictions. Now, we run a similar analysis but enable ab-initio predictions through augustus.</p>
<h2 id="prepare-the-input-data">Prepare the input data</h2>
<p>No need to re-compute the mapping/alignment of the different lines of evidence. Indeed, this time consuming task has already been performed during the previous round of annotation (evidence based). So, we will use the corresponding gff files previously produced by MAKER.</p>
<p>Link the gff files you want to use into your folder:</p>
<ul>
<li>repeatmasker.chr4.gff (already present)</li>
<li>repeatrunner.chr4.gff (already present)</li>
<li>4.fa (already present)</li>
<li>est_gff_stringtie.gff (transcript that have been mapped by MAKER during the evidence based round of annotation) </li>
<li>est2genome.gff </li>
<li>protein2genome.gff </li>
</ul>
<pre><code>ln -s maker_no_abinitio/annotationByType/est_gff:stringtie.gff est_gff_stringtie.gff
ln -s maker_no_abinitio/annotationByType/est2genome.gff 
ln -s maker_no_abinitio/annotationByType/protein2genome.gff
</code></pre>

<p>This time, we do specify a reference species to be used by augustus, which will enable ab-initio gene finding and keep_preds=1 will also show abinitio prediction not supported by any evidences :</p>
<p><em>augustus_species=fly</em> #Augustus gene prediction species model  (this is where you can call the database you trained for augustus)
...<br />
<i>protein2genome=0</i><br />
<i>est2genome=0</i></p>
<p><em>keep_preds=1</em></p>
<p>With these settings, Maker will run augustus to predict gene loci, but inform these predictions with information from the protein and est alignments.</p>
<p>Before running MAKER you can check you have modified the maker_opts.ctl file properly <a href="../practical2_supl3_maker/">here</a>.</p>
<h2 id="run-maker-with-ab-initio-predictions">Run Maker with ab-initio predictions</h2>
<p>With everything configured, run Maker as you did for the previous analysis:</p>
<pre><code>mpiexec -n 8 maker
</code></pre>

<p>We probably expect this to take a little bit longer than before, since we have added another step to our analysis.</p>
<h2 id="compile-the-output">Compile the output</h2>
<p>When Maker has finished, compile the output:</p>
<pre><code>maker_merge_outputs_from_datastore.pl --output maker_with_abinitio 
</code></pre>

<p>And again, it is probably best to link the resulting output (maker.gff) to a result folder (the same as defined in the previous exercise e.g. dmel_results), under a descriptive name.</p>
<h2 id="inspect-the-gene-models">Inspect the gene models</h2>
<p>To get some statistics of your annotation you could launch :</p>
<pre><code>gff3_sp_statistics.pl --gff maker_with_abinitio/annotationByType/maker.gff
</code></pre>

<p>We could now also visualise the annotation in the Webapollo genome browser.</p></div>
        
        
    </div>

    <footer class="col-md-12 text-center">
        
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
        
        
    </footer>

    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <!-- <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script> -->
    <script src="../../../js/base.js"></script>
    <script src="../../../search/main.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
