<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">   
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    <title></title>

    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/2.018/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/base.css" rel="stylesheet">
    <link href="../../../css/cinder.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">


    <link href="../../../css/admonition.css" rel="stylesheet">


    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 1 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 2 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../software/">Installing Software</a>
</li>

                        
                            
<li >
    <a href="../../../git/">Git</a>
</li>

                        
                            
<li >
    <a href="../../../blast/">Command-line Blast</a>
</li>

                        
                            
<li >
    <a href="../../../project_organisation/">Project Organisation</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../rna/">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../16S/">Metabarcoding</a>
</li>

                        
                            
<li class="active">
    <a href="./">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../wms/">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fa fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../16S/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../wms/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#metagenome-assembly-and-binning">Metagenome assembly and binning</a></li>
            <li class="second-level"><a href="#getting-the-data">Getting the Data</a></li>
                
            <li class="second-level"><a href="#quality-control">Quality Control</a></li>
                
            <li class="second-level"><a href="#assembly">Assembly</a></li>
                
            <li class="second-level"><a href="#binning">Binning</a></li>
                
            <li class="second-level"><a href="#checking-the-quality-of-the-bins">Checking the quality of the bins</a></li>
                
            <li class="second-level"><a href="#further-reading">Further reading</a></li>
                
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="metagenome-assembly-and-binning">Metagenome assembly and binning</h1>
<p>In this tutorial you'll learn how to inspect assemble metagenomic data and retrieve draft genomes from assembled metagenomes</p>
<p>We'll use a mock community of 20 bacteria sequenced using the Illumina HiSeq.
In reality the data were simulated using <a href="http://insilicoseq.readthedocs.io">InSilicoSeq</a>.</p>
<p>The 20 bacteria in the dataset were selected from the <a href="http://ocean-microbiome.embl.de/companion.html">Tara Ocean study</a> that recovered 957 distinct Metagenome-assembled-genomes (or MAGs) that were previsouly unknown! (full list on <a href="https://figshare.com/articles/TARA-NON-REDUNDANT-MAGs/4902923/1">figshare</a> )</p>
<h2 id="getting-the-data">Getting the Data</h2>
<pre><code class="bash">mkdir -p ~/data
cd ~/data
curl -O -J -L https://osf.io/th9z6/download
curl -O -J -L https://osf.io/k6vme/download
chmod -w tara_reads_R*
</code></pre>

<h2 id="quality-control">Quality Control</h2>
<p>we'll use FastQC to check the quality of our data, as well as sickle for trimming the bad quality part of the reads.
If you need a refresher on how and why to check the quality of sequence data, please check the <a href="../qc">Quality Control and Trimming</a> tutorial</p>
<pre><code class="bash">mkdir -p ~/results
cd ~/results
ln -s ~/data/tara_reads_* .
fastqc tara_reads_*.fastq.gz
</code></pre>

<div class="admonition question">
<p class="admonition-title">Question</p>
<p>What is the average read length? The average quality?</p>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Compared to single genome sequencing, what graphs differ?</p>
</div>
<p>Now we'll trim the reads using sickle</p>
<pre><code>sickle pe -f tara_reads_R1.fastq.gz -r tara_reads_R2.fastq.gz -t sanger \
    -o tara_trimmed_R1.fastq -p tara_trimmed_R2.fastq -s /dev/null
</code></pre>

<div class="admonition question">
<p class="admonition-title">Question</p>
<p>How many reads were trimmed?</p>
</div>
<h2 id="assembly">Assembly</h2>
<p>Megahit will be used for the assembly.</p>
<pre><code>megahit -1 tara_trimmed_R1.fastq -2 tara_trimmed_R2.fastq -o tara_assembly
</code></pre>

<p>the resulting assenmbly can be found under <code>tara_assembly/final.contigs.fa</code>.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>How many contigs does this assembly contain?</p>
</div>
<h2 id="binning">Binning</h2>
<p>First we need to map the reads back against the assembly to get coverage information</p>
<pre><code class="bash">ln -s tara_assembly/final.contigs.fa .
bowtie2-build final.contigs.fa final.contigs
bowtie2 -x final.contigs -1 tara_reads_R1.fastq.gz -2 tara_reads_R2.fastq.gz | \
    samtools view -bS -o tara_to_sort.bam
samtools sort tara_to_sort.bam -o tara.bam
samtools index tara.bam
</code></pre>

<p>then we run metabat</p>
<pre><code class="bash">runMetaBat.sh -m 1500 final.contigs.fa tara.bam
mv final.contigs.fa.metabat-bins1500 metabat
</code></pre>

<div class="admonition question">
<p class="admonition-title">Question</p>
<p>How many bins did we obtain?</p>
</div>
<h2 id="checking-the-quality-of-the-bins">Checking the quality of the bins</h2>
<p>The first time you run <code>checkm</code> you have to create the database</p>
<pre><code class="bash">sudo checkm data setRoot ~/.local/data/checkm
</code></pre>

<pre><code class="bash">checkm lineage_wf -x fa metabat checkm/
checkm bin_qa_plot -x fa checkm metabat plots
</code></pre>

<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Which bins should we keep for downstream analysis?</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>checkm can plot a lot of metrics. If you have time, check the manual
and try to produce different plots</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>if checkm fails at the phylogeny step, it is likely that your vm doesn't have enough RAM.
pplacer requires about 35G of RAM to place the bins in the tree of life.</p>
<p>In that case, execute the following</p>
<p><code>cd ~/results</code><br />
<code>curl -O -J -L https://osf.io/xuzhn/download</code><br />
<code>tar xzf checkm.tar.gz</code><br />
<code>checkm qa checkm/lineage.ms checkm</code>  </p>
</div>
<p>then plot the completeness</p>
<pre><code class="bash">checkm bin_qa_plot -x fa checkm metabat plots
</code></pre>

<p>and take a look at <code>plots/bin_qa_plot.png</code></p>
<h2 id="further-reading">Further reading</h2>
<ul>
<li><a href="https://www.nature.com/articles/s41564-017-0012-7">Recovery of nearly 8,000 metagenome-assembled genomes substantially expands the tree of life</a></li>
<li><a href="https://www.nature.com/articles/sdata2017203">The reconstruction of 2,631 draft metagenome-assembled genomes from the global oceans</a></li>
</ul></div>
        
    </div>

    <footer class="col-md-12 text-center">
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
    </footer>

    <script src="../../../js/jquery-1.10.2.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
    <script src="../../../js/base.js"></script>
    <script src="../../../search/require.js"></script>
    <script src="../../../search/search.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
